security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisTool.java

```diff
@@ -155,23 +159,36 @@ protected TableFormatterConfig createTableFormatterConfig() {
         return line.hasOption(option) ? Optional.of(line.getOptionValue(option)) : Optional.empty();
     }
 
+    private static void copyBytesToPath(byte[] bytes, Path dest) {
+        try {
+            Files.write(dest, bytes);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
     private static SecurityAnalysisResult runSecurityAnalysis(CommandLine line, ToolRunningContext context, ContingenciesProvider contingenciesProvider, SecurityAnalysisParameters parameters, SecurityAnalysis securityAnalysis, String currentState) {
         SecurityAnalysisResult result;
         if (!line.hasOption(OUTPUT_LOG_OPTION)) {
             result = securityAnalysis.run(currentState, parameters, contingenciesProvider).join();
         } else {
-            SecurityAnalysisResultWithLog resultWithLog = securityAnalysis.runWithLog(currentState, parameters, contingenciesProvider).join();
-            result = resultWithLog.getResult();
-            // copy log bytes to file
-            resultWithLog.getLogBytes().ifPresent(logBytes -> {
-                Path outlogDest = context.getFileSystem().getPath(line.getOptionValue(OUTPUT_LOG_OPTION));
-                try (ByteArrayInputStream bis = new ByteArrayInputStream(logBytes);
-                     OutputStream fos = Files.newOutputStream(outlogDest)) {
-                    IOUtils.copy(bis, fos);
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
+            try {
+                SecurityAnalysisResultWithLog resultWithLog = securityAnalysis.runWithLog(currentState, parameters, contingenciesProvider).join();
+                result = resultWithLog.getResult();
+                // copy log bytes to file
+                resultWithLog.getLogBytes().ifPresent(logBytes -> {
+                    Path outlogDest = context.getFileSystem().getPath(line.getOptionValue(OUTPUT_LOG_OPTION));
+                    copyBytesToPath(logBytes, outlogDest);
+                });
+            } catch (CompletionException e) {
+                if (e.getCause() instanceof ComputationException) {
+                    ComputationException computationException = (ComputationException) e.getCause();
+                    byte[] bytes = computationException.toZipBytes();
+                    Path outlogDest = context.getFileSystem().getPath(line.getOptionValue(OUTPUT_LOG_OPTION));
```

Comment:
If the OUTPUT_LOG_OPTION is required? I think adding a required option should be considered as a breaking change

Reply:
Not required.